### DISCUSSION TITLE
on `next/image`, hostname is not configured under images in your `next.config.js`

### DISCUSSION THREAD

**Discussion by Vince-vegas** [vercel/next.js#20953] [Help] [answered]
Why hostname error keep showing on the page even though I've implemented **next.config.js** on my  projects with domains on it.

**The Config file**
```
module.exports = {
  images: {
    domains: ['https://image.tmdb.org/', 'https://api.themoviedb.org/'],
  },
};

```

// Page
```
import React from 'react';
import Image from 'next/image';

const BrowseMovie = ({ browseMovie }) => {
  return (
    <div>
      <h1>Movies</h1>
      <div className="d-flex">
        {browseMovie.map((item) => {
          return (
            <div key={item.id} className="card-movie">
              <Image
                src={`https://image.tmdb.org/t/p/w300${item.poster_path}`}
                height={300}
                width={300}
                alt={item.title}
              />
            </div>
          );
        })}
      </div>
    </div>
  );
};

export async function getStaticProps() {
  const movies = await fetch(
    `https://api.themoviedb.org/3/discover/movie?api_key=******&language=en-US&sort_by=popularity.desc&include_adult=false&include_video=false&page=1`
  );
  const data = await movies.json();

  return {
    props: {
      browseMovie: data.results,
    },
  };
}

export default BrowseMovie;

```

**Comment by jamesmosier** [comment, marked as answer]
Try setting up your domains like this:

```js
module.exports = {
  images: {
    domains: ['tmdb.org', 'themoviedb.org'],
  },
};
```

Only the actual domain should need setup without the protocol and any subdomains.

**Reply by Vince-vegas** [reply]
When I added the domain **image.tmdb.org**. The error  is now solved.

**Reply by Besuf** [reply]
It's tricky you shouldn't add https:// in your domain

**Reply by DuudeXX8** [reply]
Do you already solve it?

**Reply by skyespirates** [reply]
yeey, it's worked for me thanks

**Reply by ventura01** [reply]
Great!

**Reply by helviorhild** [reply]
follow instruction, and at the end restart your dev session. (don't forget your friend "ctrl+c" )

**Comment by verhulstd** [comment]
I am having the same issue..
This is my next.config.js in root of project
```
module.exports = {
  images: {
    domains: ["image.tmdb.org"],
  },
};
```

but still giving me this error:
```
Server Error
Error: Invalid src prop (https://image.tmdb.org/t/p/w500/dAvJHKwXGWsTNCgC3nCiqGOnagx.jpg) on `next/image`, hostname "image.tmdb.org" is not configured under images in your `next.config.js`
See more info: https://nextjs.org/docs/messages/next-image-unconfigured-host
```

**Reply by mrlaseptima** [reply]
same  :(

**Reply by katungi** [reply]
I also had this issue, Did you restart your server after editing the `next.config` file?

**Reply by fer0o** [reply]
I restarted the server and work fine. Thanks man

**Reply by adamaslan** [reply]
restarted the server and its still messed up :/

**Reply by rohilpinto** [reply]
make sure to add width & height props or layout ='fill'

**Comment by wellitongervickas** [comment]
Works for me: https://github.com/vercel/next.js/issues/18412#issuecomment-801708220
Create your own image loader maybe help you

**Reply by wellitongervickas** [reply]
```
import Image from "next/image";

const externaImageLoader = ({ src }: { src: string }) =>
  `https://static2.taglivros.com/${src}`;

const Book = ({ book }: { book: Books.Book }) => (
  <div>
    <Image
      loader={externaImageLoader}
      src={book.cover.name}
      alt={book.name}
      layout="fill"
    />
  </div>
);

export default Book;

```

**Reply by idemmanuel001** [reply]
This worked for me, thank you.

**Comment by sanjayrjs16** [comment]
I've got a question here. 
What if the `src` comes out to be something not in` next.config.js` where the domain name is only know at run time ?
Eg: User has option to upload any image to the app (or paste any link) , and src would be known only at run time.

How do we tackle this issue ?

**Reply by arthurfiorette** [reply]
I think at the moment the only way is to disable image optimization.

```tsx
<Image 
  unoptimized 
  src="..."
  alt="..."
  // ...
/>
```

**Reply by vladimirrostok** [reply]
That helped me very much, thank you for bringing this up!

**Reply by sxanni** [reply]
Th

> I think at the moment the only way is to disable image optimization.
> 
> ```tsx
> <Image 
>   unoptimized 
>   src="..."
>   alt="..."
>   // ...
> />
> ```

Thankssssssssssss

**Reply by janlin002** [reply]
Thankssssssssssss

**Comment by EliseiNicolae** [comment]
For those who still don't work, try to add on `next.config.js`:
```
module.exports = {
  images: {
    domains: 'api-production.s3.amazonaws.com', 'Your_domain'],
  },
}
```

**And then run the app again.** 📦

**Reply by ghost** [reply]
OOOOOOOOh thank you a looooooot !!!!!!

**Reply by ghost** [reply]
Do you know why we have to add this ?

**Reply by EliseiNicolae** [reply]
Adding the domains in the images configuration is important because by default, Next.js only allows images to be loaded from the same domain as the application itself. This is a security feature to prevent malicious code from loading images from untrusted sources.

However, in some cases, you may want to load images from a different domain, such as when you're using a content delivery network (CDN) to serve your images. By adding the allowed domains to the images configuration, you're telling Next.js that these domains are trusted sources for images, and it should allow images to be loaded from them.

**Reply by sahandghorbani** [reply]
very helpful

**Comment by sgehrman** [comment]
This is pitiful.  Why did I choose NextJS???  So stupid.

**Comment by aceaspades-worldspark** [comment]
Is there any way to use wildcard domains, without setting ```unoptimized```?

**Reply by natromankevich** [reply]
Yes, here https://github.com/vercel/next.js/discussions/20953#discussioncomment-918141

**Comment by Kamasah-Dickson** [comment]
Hello guys I really need some help here. I am creating a podcast application using the taddy api.  I have created a search page that allows users to search for any podcast but it seems like the images have different hostnames and there are already thousands of podcasts with different hostnames. I don't know how  to fix this and I can't add all those hostnames.

**Reply by Psiale** [reply]
Seems like you can disable optimization by calling the `unoptimized` property on the `Image` component

**Reply by Kamasah-Dickson** [reply]
I was able to go about it by adding some hostnames i wanted to inlcude

```js
/** @type {import('next').NextConfig} */
const nextConfig = {
	reactStrictMode: true,
	images: {
		domains: [
			"d3t3ozftmdmh3i.cloudfront.net",
			"s3.amazonaws.com",
			"afr.net",
			"ott.bgea.live",
			"deow9bq0xqvbj.cloudfront.net",
			"d3wo5wojvuv7l.cloudfront.net",
			"episodes.castos.com",
		],
		remotePatterns: [
			{
				protocol: "http",
				hostname: "**",
			},
			{
				protocol: "https",
				hostname: "**",
			},
		],
	},
};

module.exports = nextConfig;
```

**Reply by idityaGE** [reply]
thanks its was a great help

**Reply by reply2sagar** [reply]
Can someone explain how to skip this error. If image host is not configured in config file, error should not be thrown. Instead it should show blank image so that entire app does not crash.

**Comment by reply2sagar** [comment]
Can someone explain how to skip this error. If image host is not configured in config file, error should not be thrown. Instead it should show blank image so that entire app does not crash.

**Comment by Mnigos** [comment]
I'm also facing that issue
